modules/epang/main.nf

```diff
@@ -0,0 +1,57 @@
+process EPANG {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::epa-ng=0.3.8" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/epa-ng:0.3.8--h9a82719_1':
+        'quay.io/biocontainers/epa-ng:0.3.8--h9a82719_1' }"
+
+    input:
+    tuple val(meta), path(queryaln)
+    path referencealn
+    path referencetree
+    path bfastfile
+    path splitfile
+    path binaryfile
+
+    output:
+    tuple val(meta), path("./."), emit: epang   , optional: true
+    path "*.epa_result.jplace"  , emit: jplace  , optional: true
+    path "*.epa_info.log"       , emit: log
+    path "versions.yml"         , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args       = task.ext.args   ?: ''
+    def prefix     = task.ext.prefix ?: "${meta.id}"
+    def queryarg   = queryaln        ? "--query $queryaln"       : ""
+    def refalnarg  = referencealn    ? "--ref-msa $referencealn" : ""
+    def reftreearg = referencetree   ? "--tree $referencetree"   : ""
+    def modelarg   = meta.model      ? "--model $meta.model"     : ""
```

Comment:
I just remembered, you don't need this at all. You can do this entirely with `ext.args` in `modules.conf`! this way you don't have to 'assume' a `meta` entry at all!

```
    ext.args = { "--model ${meta.model}" }
```

This way you don't have to assume anything and reduce the number of input channels.

We do this in taxprofiler: https://github.com/nf-core/taxprofiler/blob/6f87c681fd455a30594b2fb37f8d42e7f668b067/conf/modules.config#L267